apps/cpp_rpc/rpc_env.cc

```diff
@@ -95,7 +95,13 @@ RPCEnv::RPCEnv(const std::string& wd) {
     auto cmdline = fopen("/proc/self/cmdline", "r");
     fread(cwd, 1, sizeof(cwd), cmdline);
     fclose(cmdline);
-    base_ = "/data/data/" + std::string(cwd) + "/cache/rpc";
+    std::string android_base_ = "/data/data/" + std::string(cwd) +"/cache";
+    struct stat statbuf;
+    if (stat(android_base_.data(), &statbuf) == -1 || !S_ISDIR(statbuf.st_mode)){
+        android_base_ = ".";
```

Comment:
If failed, android_base_ becomes `.`,  will `./rpc` work well on android?

Reply:
When current working directory is writable, Yes .

Another option is force `base_` to `/data/local/tmp/rpc`. But I don't think this is a good choice. So I keep the original implementation .

Reply:
how about add one simple code comment to indicate users if failed, should make sure the directory is writable and also consider to change `base_` to `/data/local/tmp/rpc`. This could help users find one solution more quickly